### PR TITLE
Expose a new .__parse() method for generating a parse tree from a math expression

### PR DESCRIPTION
**Pull request by jeresig**
Explicitly marked with tons of warnings, both in name and in comments. To be used for some internal projects, for the time being. Added a simple test, just as a sanity check.

### CODE REVIEW

File: test/katex-spec.js [hunk 1]
```diff
@@ -1116,6 +1116,31 @@ describe("A markup generator", function() {
     });
 });
 
+describe("A parse tree generator", function() {
+    it("generates a tree", function() {
+        var tree = katex.__parse("\\sigma^2");
+        expect(JSON.stringify(tree)).toEqual(JSON.stringify([
```
Comment (jeresig) [inline test/katex-spec.js]: Surprisingly just doing `expect(tree).toEqual(...)` failed! Doing a string comparison seemed to work, though!

**Comment by jeresig** [comment]
cc: @ariabuckles @xymostech

File: test/katex-spec.js (hunk 1 above)
Comment (xymostech) [inline test/katex-spec.js]: I think that `toEqual` just does an `===` compare. You could try `expect(_.isEqual(tree, ...)).toBe(true)`?

Comment (xymostech) [inline test/katex-spec.js]: No, never mind, `toEqual` does do deep compare: [jasmine's deep `eq` function](https://github.com/jasmine/jasmine/blob/33641578e6f7d90be6155ea6dcb262b11a326fea/src/core/matchers/matchersUtil.js#L53). The only thing I can think of is that `"sub": undefined` should be `"sub": null`?

Comment (jeresig) [inline test/katex-spec.js]: Yeah - that was my thought as well. The method generates a `"sub": undefined`. I tried leaving it out and adding it in -- it failed in both cases. Not sure :(

**Comment by jeresig** [comment]
I'm going to merge this so that I can finish up my react-components changes -- let me know if you want me to make any additional changes to the tests!

**Comment by xymostech** [comment]
Oops, forgot this hadn't gotten merged in. Looks good to me! If the tests break at some point or something, I'll maybe change them, but if they're working locally and on Travis then :+1:

**Pull request merged**